lib/http_perform.c

```diff
@@ -131,10 +131,13 @@ sf_bool STDCALL http_perform(CURL *curl,
                              int64 network_timeout,
                              sf_bool chunk_downloader,
                              SF_ERROR_STRUCT *error,
-                             sf_bool insecure_mode) {
+                             sf_bool insecure_mode,
+                             sf_bool retry_on_curle_couldnt_connect,
+                             int8 retry_on_curle_couldnt_connect_count) {
```

Comment:
This is a transient failure and retry logic should work, It would be better if retry context is used with exponential backoff and sleep with jitter is used.